src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -209,4 +250,20 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
             filterToAnsweredOrReacted: false,
         });
     }
+
+    /**
+     * helper method which returns the order which posts must be listed
+     * @param postA     first post to compare
+     * @param postB     second post to compare
+     * @return number   the order which posts must be listed
+     */
+    sortByDate = (postA: Post, postB: Post): number => {
+        if (Number(postA.creationDate) > Number(postB.creationDate)) {
+            return this.currentSortDirection === SortDirection.DESCENDING ? -1 : 1;
+        }
+        if (Number(postA.creationDate) < Number(postB.creationDate)) {
+            return this.currentSortDirection === SortDirection.DESCENDING ? 1 : -1;
+        }
+        return 0;
```

Comment:
This method would treat two posts as equal if either `postA.creationDate` or `postB.creationDate` is `undefined` (`Number(undefined)` is `NaN` and comparison with `NaN` always gives false). I wanted to be sure if this is intended or should be handled as well?

Reply:
Thanks for your review. In the current context with checks both in the client and the server, we never have this condition so we should not worry about it. Postings cannot exist without creation date.